src/machine_charm.py

```diff
@@ -147,14 +147,15 @@ class GrafanaAgentMachineCharm(GrafanaAgentCharm):
 
     service_name = "grafana-agent.grafana-agent"
 
-    # Pairs of (incoming, outgoing) relation names. If any 'incoming' is joined without a matching
-    # 'outgoing', the charm will block. Without an outgoing relation we may incur data loss.
+    # Pairs of (incoming, [outgoing]) relation names. If any 'incoming' is joined
+    # without a matching 'outgoing', the charm will block. Without an outgoing relation
+    # we may incur data loss.
     mandatory_relation_pairs = [
-        ("cos-agent", "send-remote-write"),
-        ("cos-agent", "logging-consumer"),
-        ("cos-agent", "grafana-dashboards-provider"),
-        ("juju-info", "send-remote-write"),
-        ("juju-info", "logging-consumer"),
+        ("cos-agent", ["send-remote-write", "grafana-cloud-config"]),
+        ("cos-agent", ["logging-consumer", "grafana-cloud-config"]),
+        ("cos-agent", ["grafana-dashboards-provider"]),
+        ("juju-info", ["send-remote-write", "grafana-cloud-config"]),
+        ("juju-info", ["logging-consumer", "grafana-cloud-config"]),
```

Comment:
I know this isn't a bug in this change but this seems like the best time to fix it.

We do not want:
```
("cos-agent", ["send-remote-write", "grafana-cloud-config"]),
("cos-agent", ["logging-consumer", "grafana-cloud-config"]),
("cos-agent", ["grafana-dashboards-provider"]),
```
We instead want:
```
("cos-agent", ["send-remote-write", "logging-consumer", "grafana-cloud-config"])
```

This is because using cos-agent for only logs or only metrics is a completely fine use case. Same goes for juju-info.